cmd/collector/app/handler.go

```diff
@@ -80,6 +85,51 @@ func (aH *APIHandler) saveSpan(w http.ResponseWriter, r *http.Request) {
 		} else {
 			w.WriteHeader(http.StatusOK)
 		}
+
+	case "zipkin.thrift":
+		bodyBytes, err := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		if err != nil {
+			http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+
+		buffer := thrift.NewTMemoryBuffer()
+		if _, err = buffer.Write(bodyBytes); err != nil {
+			http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+
+		transport := thrift.NewTBinaryProtocolTransport(buffer)
+		_, size, err := transport.ReadListBegin()
```

Comment:
I don't like this hack because we would need to create a copy of the byte array, pad in the front and the back. I didn't realize that the statements weren't covered, I've added tests.